libvips/foreign/foreign.c

```diff
@@ -1718,26 +1718,30 @@ vips_foreign_save_remove_metadata(VipsImage *image,
 {
 	VipsForeignKeep keep = *((VipsForeignKeep *) user_data);
 
-	// we are only interested in metadata
-	if (!vips_isprefix("png-comment-", field) &&
-		!vips_isprefix("magickprofile-", field) &&
-		strcmp(field, VIPS_META_IMAGEDESCRIPTION) != 0 &&
-		!g_str_has_suffix(field, "-data"))
-		return NULL;
-
-	if ((strcmp(field, VIPS_META_EXIF_NAME) == 0 &&
+	if ((g_str_equal(field, VIPS_META_EXIF_NAME) &&
 			(keep & VIPS_FOREIGN_KEEP_EXIF)) ||
-		(strcmp(field, VIPS_META_XMP_NAME) == 0 &&
+		(g_str_equal(field, VIPS_META_XMP_NAME) &&
 			(keep & VIPS_FOREIGN_KEEP_XMP)) ||
-		(strcmp(field, VIPS_META_IPTC_NAME) == 0 &&
+		(g_str_equal(field, VIPS_META_IPTC_NAME) &&
 			(keep & VIPS_FOREIGN_KEEP_IPTC)) ||
-		(strcmp(field, VIPS_META_ICC_NAME) == 0 &&
+		(g_str_equal(field, VIPS_META_ICC_NAME) &&
 			(keep & VIPS_FOREIGN_KEEP_ICC)) ||
-		(keep & VIPS_FOREIGN_KEEP_OTHER))
+		(vips_isprefix("gainmap", field) &&
+			(keep & VIPS_FOREIGN_KEEP_GAINMAP)))
+		return NULL;
+
+	/* OTHER is everything that doesn't match one of the above tests.
+	 */
```

Comment:
I think(?) the original intent of that old check was to define an allow list of which metadata should be classified as "other".

For example, when saving with `keep=none`, this would now also remove `"delay"`, `"loop"`, `"resolution-unit"`, and others (grep for `vips_image_get*` in the savers), which affect saver behavior rather than representing metadata within the image.